caddyhttp/httpserver/plugin.go

```diff
@@ -50,6 +50,11 @@ func init() {
 	caddy.RegisterServerType(serverType, caddy.ServerType{
 		Directives: func() []string { return directives },
 		DefaultInput: func() caddy.Input {
+			if Port == DefaultHTTPSPort && serverType == "http" {
```

Comment:
>  I was under the assumption that raising an error and exiting would be the valid fix.

But this is not how to raise the error. I think reading the godoc to learn about Caddy's architecture would help. :) But again, Caddy 2 is around the corner, so Caddy 1 is becoming less relevant.